framework/test/mocha/unit/modules/http_api/controllers/node.js

```diff
@@ -194,6 +200,214 @@ describe('node/api', () => {
+					},
+					{
+						height: 457,
+					},
+					{
+						height: MAJORITY_HEIGHT,
+					},
+					{
+						height: 453,
+					},
+					{
+						height: MAJORITY_HEIGHT,
+					},
+					{
+						height: 438,
+					},
+				];
+				channelStub.invoke
+					.withArgs('network:getConnectedPeers')
+					.returns(defaultPeers);
+				networkHeight = await getNetworkHeight();
+			});
+
+			it('should return correct networkHeight based on majority', async () => {
+				expect(networkHeight).to.be.eql(MAJORITY_HEIGHT);
+			});
+		});
+
+		describe('When network:getConnectedPeers returns a blank list of peers', () => {
+			beforeEach(async () => {
+				channelStub.invoke.withArgs('network:getConnectedPeers').returns([]);
+				networkHeight = await getNetworkHeight();
+			});
+
+			it('should return zero when there are no peers with height', async () => {
+				expect(networkHeight).to.be.eql(0);
+			});
+		});
+
+		describe('When network:getConnectedPeers returns a list of peers with 2 different equal set of peers height', () => {
+			beforeEach(async () => {
+				const defaultPeers = [
+					{
+						height: MAJORITY_HEIGHT + 1,
+					},
+					{
+						height: MAJORITY_HEIGHT + 1,
+					},
+					{
+						height: MAJORITY_HEIGHT + 1,
+					},
+					{
+						height: MAJORITY_HEIGHT,
+					},
+					{
+						height: MAJORITY_HEIGHT,
+					},
+					{
+						height: MAJORITY_HEIGHT,
+					},
+				];
+				channelStub.invoke
+					.withArgs('network:getConnectedPeers')
+					.returns(defaultPeers);
+				networkHeight = await getNetworkHeight();
+			});
+
+			it('should return height of majority with lower height', async () => {
+				expect(networkHeight).to.be.eql(MAJORITY_HEIGHT);
+			});
+		});
+
+		describe('When network:getConnectedPeers returns a list of peers with 3 different equal set of peers height', () => {
+			beforeEach(async () => {
+				const defaultPeers = [
+					{
+						height: MAJORITY_HEIGHT + 2,
+					},
+					{
+						height: MAJORITY_HEIGHT + 2,
+					},
+					{
+						height: MAJORITY_HEIGHT + 2,
+					},
+					{
+						height: MAJORITY_HEIGHT + 1,
+					},
+					{
+						height: MAJORITY_HEIGHT + 1,
+					},
+					{
+						height: MAJORITY_HEIGHT + 1,
+					},
+					{
+						height: MAJORITY_HEIGHT,
+					},
+					{
+						height: MAJORITY_HEIGHT,
+					},
+					{
+						height: MAJORITY_HEIGHT,
+					},
+				];
+				channelStub.invoke
+					.withArgs('network:getConnectedPeers')
+					.returns(defaultPeers);
+				networkHeight = await getNetworkHeight();
+			});
+
+			it('should return height of majority with lower height', async () => {
+				expect(networkHeight).to.be.eql(MAJORITY_HEIGHT);
+			});
+		});
+
+		describe('When network:getConnectedPeers returns a list of peers with 2 different unequal set of peers height', () => {
+			beforeEach(async () => {
+				const defaultPeers = [
+					{
+						height: MAJORITY_HEIGHT,
+					},
+					{
+						height: MAJORITY_HEIGHT,
+					},
+					{
+						height: MAJORITY_HEIGHT + 1,
```

Comment:
`MAJORITY_HEIGHT` here and `MAJORITY_HEIGHT - 1` makes more sense, but this is fine also.